echopype/echodata/combine.py

```diff
@@ -218,6 +221,17 @@ def combine_echodata(echodatas: List[EchoData], combine_attrs="override") -> Ech
                         new_time2 = combined_group["time2"]
                     else:
                         combined_group["time2"] = new_time2
+                if "time3" in combined_group and exist_reversed_time(combined_group, "time3"):
+                    if old_time3 is None:
+                        warnings.warn(
+                            f"{sonar_model} time3 reversal detected; the times will be corrected"  # noqa
+                            " (see https://github.com/OSOceanAcoustics/echopype/pull/297)"
+                        )
+                        old_time3 = combined_group["time3"]
+                        coerce_increasing_time(combined_group, time_name="time3")
+                        new_time3 = combined_group["time3"]
+                    else:
+                        combined_group["time3"] = new_time3
```

Comment:
I think the highlighted repeated block can be combined into a single function and called for each of the `ping_time`/`time1/2/3` cases because all of the operations and printouts are identical, except for the variable name, and the check for `nmea` is _before_ this identical block also.